File: src/FizzBuzz Variety Sample Codes/fizzbuzz.exs
# モジュールを定義 --- (*1)
defmodule FizzBuzz do
  # FizzBuzzを返す関数を定義 --- (*2)
  def getValue(i) do
    cond do
      Integer.mod(i, 15) == 0 -> "FizzBuzz"
      Integer.mod(i, 3)  == 0 -> "Fizz"
      Integer.mod(i, 5)  == 0 -> "Buzz"
      :else                   -> i
    end
  end
end
# 100回繰り返す --- (*3)
for i <- 1..100 do
  IO.puts FizzBuzz.getValue(i)
end

 

File: src/FizzBuzz Variety Sample Codes/fizzbuzz.rb
# FizzBuzzに応じた値を返す関数 --- (*1)
def fizzbuzz(i)
  if i % 3 == 0 and i % 5 == 0
    "FizzBuzz"
  elsif i % 3 == 0
    "Fizz"
  elsif i % 5 == 0
    "Buzz"
  else
    i
  end
end

# 1から100まで繰り返しfizzbuzzを呼ぶ --- (*2)
1.upto(100) do |i|
  puts fizzbuzz(i)
end

